### ISSUE TITLE
Insert table exception influence next query

### ISSUE THREAD

**Issue by hexiaoting** [ClickHouse/ClickHouse#16384]
**Describe the bug**
If the first insert query make an exception, then next query will return 0 rows.

**How to reproduce**
```
dell123 :) show tables;

SHOW TABLES

Query id: c23a8058-009d-4e6c-b8f1-0f4e6eb198e9

┌─name─────────────────┐
│ array                │
│ columns_transformers │
│ json                 │
│ json1                │
│ t1                   │
└──────────────────────┘

5 rows in set. Elapsed: 0.003 sec.

dell123 :) insert into array values([1+1, 2], 'aaa');

INSERT INTO array VALUES

Query id: 2a26e56c-25ab-4e8b-99b1-dec662c3ec9d

Error on processing query: insert into array values([1+1, 2], 'aaa');
Code: 62, e.displayText() = DB::Exception: parse failed 0: data for INSERT was parsed from query, Stack trace (when copying this message, always include the lines below):

0. /mnt/disk4/hewenting/ClickHouse/src/contrib/poco/Foundation/src/Exception.cpp:27: Poco::Exception::Exception(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int) @ 0x1189d3 in /mnt/disk4/hewenting/ClickHouse/build-dev/contrib/poco-cmake/Foundation/lib_poco_foundation.so
1. /mnt/disk4/hewenting/ClickHouse/src/src/Common/Exception.cpp:38: DB::Exception::Exception(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int) @ 0x19ae19 in /mnt/disk4/hewenting/ClickHouse/build-dev/src/libclickhouse_common_io.so
2. /mnt/disk4/hewenting/ClickHouse/src/contrib/libcxx/include/string:2134: DB::ValuesBlockInputFormat::generate() @ 0x2ba9f1 in /mnt/disk4/hewenting/ClickHouse/build-dev/src/libclickhouse_processors_formats_impl.so
3. /mnt/disk4/hewenting/ClickHouse/src/contrib/libcxx/include/vector:1003: DB::ISource::work() @ 0x10cac7 in /mnt/disk4/hewenting/ClickHouse/build-dev/src/libclickhouse_processors.so
4. /mnt/disk4/hewenting/ClickHouse/src/src/Processors/Formats/InputStreamFromInputFormat.h:49: DB::InputStreamFromInputFormat::readImpl() @ 0x1a8fdd in /mnt/disk4/hewenting/ClickHouse/build-dev/src/libclickhouse_datastreams.so
5. /mnt/disk4/hewenting/ClickHouse/src/contrib/libcxx/include/vector:1003: DB::IBlockInputStream::read() @ 0x1755e3 in /mnt/disk4/hewenting/ClickHouse/build-dev/src/libclickhouse_datastreams.so
6. /mnt/disk4/hewenting/ClickHouse/src/contrib/libcxx/include/vector:1003: DB::AsynchronousBlockInputStream::calculate() @ 0x14b62b in /mnt/disk4/hewenting/ClickHouse/build-dev/src/libclickhouse_datastreams.so
7. /mnt/disk4/hewenting/ClickHouse/src/contrib/libcxx/include/atomic:856: std::__1::__function::__func<DB::AsynchronousBlockInputStream::next()::'lambda'(), std::__1::allocator<DB::AsynchronousBlockInputStream::next()::'lambda'()>, void ()>::operator()() @ 0x14bac9 in /mnt/disk4/hewenting/ClickHouse/build-dev/src/libclickhouse_datastreams.so
8. /mnt/disk4/hewenting/ClickHouse/src/contrib/libcxx/include/functional:1853: ThreadPoolImpl<ThreadFromGlobalPool>::worker(std::__1::__list_iterator<ThreadFromGlobalPool, void*>) @ 0x1eb6ec in /mnt/disk4/hewenting/ClickHouse/build-dev/src/libclickhouse_common_io.so
9. /mnt/disk4/hewenting/ClickHouse/src/src/Common/ThreadPool.h:177: ThreadFromGlobalPool::ThreadFromGlobalPool<void ThreadPoolImpl<ThreadFromGlobalPool>::scheduleImpl<void>(std::__1::function<void ()>, int, std::__1::optional<unsigned long>)::'lambda1'()>(void&&, void ThreadPoolImpl<ThreadFromGlobalPool>::scheduleImpl<void>(std::__1::function<void ()>, int, std::__1::optional<unsigned long>)::'lambda1'()&&...)::'lambda'()::operator()() @ 0x1ebcd1 in /mnt/disk4/hewenting/ClickHouse/build-dev/src/libclickhouse_common_io.so
10. /mnt/disk4/hewenting/ClickHouse/src/contrib/libcxx/include/functional:1853: ThreadPoolImpl<std::__1::thread>::worker(std::__1::__list_iterator<std::__1::thread, void*>) @ 0x1eb0ac in /mnt/disk4/hewenting/ClickHouse/build-dev/src/libclickhouse_common_io.so
11. /mnt/disk4/hewenting/ClickHouse/src/contrib/libcxx/include/memory:2615: void* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void ThreadPoolImpl<std::__1::thread>::scheduleImpl<void>(std::__1::function<void ()>, int, std::__1::optional<unsigned long>)::'lambda1'()> >(void*) @ 0x1e95ff in /mnt/disk4/hewenting/ClickHouse/build-dev/src/libclickhouse_common_io.so
12. start_thread @ 0x7ef7 in /home/amos/gentoo/lib64/libpthread-2.31.so
13. __clone @ 0xf9bbf in /home/amos/gentoo/lib64/libc-2.31.so
 (version 2
dell123 :)
dell123 :)
dell123 :) show tables;

SHOW TABLES

Query id: ba4e0c06-2321-4d79-b185-5701e228c6da

Ok.

0 rows in set. Elapsed: 0.000 sec.

dell123 :)
```
* Which ClickHouse server version to use
```
┌─version()─┐
│ 20.11.1.1 │
└───────────┘
```

**Comment by alexey-milovidov** [comment]
Cannot reproduce on master.

**Comment by alexey-milovidov** [comment]
```
$ clickhouse-client 
ClickHouse client version 20.12.1.1.
Connecting to localhost:9000 as user default.
Connected to ClickHouse server version 20.12.1 revision 54442.

milovidov-desktop :) show tables;

SHOW TABLES

Query id: 5a1154d1-8dca-48dd-b2a2-4be90bf08a2e

┌─name──────────────────┐
│ a                     │
│ constCondOptimization │
│ dict2                 │
│ f                     │
│ lc_null_int8_defnull  │
│ leftjoin              │
│ other_table           │
│ queries               │
│ r1                    │
│ r2                    │
│ report                │
│ rollup_having         │
│ single_id             │
│ t                     │
│ t1                    │
│ t2                    │
│ t_dist                │
│ t_local               │
│ table1                │
│ tableClick            │
│ tableConversion       │
│ table_name            │
│ test1                 │
│ testcp                │
│ topk                  │
│ ttest                 │
│ u_perf                │
│ userid_set            │
│ userid_test           │
│ v                     │
│ v1                    │
│ v2                    │
└───────────────────────┘

32 rows in set. Elapsed: 0.002 sec. 

milovidov-desktop :) insert into a values([1+1, 2], 'aaa');

INSERT INTO a VALUES

Query id: e1afd770-38f0-4e76-aeb3-75cf12ce6583


Exception on client:
Code: 43. DB::Exception: Illegal type Array(UInt64) of argument of function toDate: while executing 'FUNCTION CAST(array(plus(_dummy_0, _dummy_1), _dummy_2), 'Date') Date = CAST(array(plus(_dummy_0, _dummy_1), _dummy_2), 'Date')': data for INSERT was parsed from query

Connecting to localhost:9000 as user default.
Connected to ClickHouse server version 20.12.1 revision 54442.

milovidov-desktop :) show tables;

SHOW TABLES

Query id: 24270a5b-6bc9-4c30-8c37-c138de87d281

┌─name──────────────────┐
│ a                     │
│ constCondOptimization │
│ dict2                 │
│ f                     │
│ lc_null_int8_defnull  │
│ leftjoin              │
│ other_table           │
│ queries               │
│ r1                    │
│ r2                    │
│ report                │
│ rollup_having         │
│ single_id             │
│ t                     │
│ t1                    │
│ t2                    │
│ t_dist                │
│ t_local               │
│ table1                │
│ tableClick            │
│ tableConversion       │
│ table_name            │
│ test1                 │
│ testcp                │
│ topk                  │
│ ttest                 │
│ u_perf                │
│ userid_set            │
│ userid_test           │
│ v                     │
│ v1                    │
│ v2                    │
└───────────────────────┘

32 rows in set. Elapsed: 0.002 sec. 

milovidov-desktop :)
```

**Comment by alexey-milovidov** [comment]
`/mnt/disk4/hewenting/`

It means that you are using your own build of ClickHouse, not the official build.

**Comment by alexey-milovidov** [comment]
`DB::Exception: parse failed 0`

We don't have this exception in our code.
PS. We always write proper exception messages, as a sentence starting from capital letter, with clear explanation.

**Comment by hexiaoting** [comment]
@alexey-milovidov I'm sorry for I forget to remove my own debug info "parse failed 0" before paste the error stack to this issue. 

But I reproduce this exception on the latest master branch:
```
show tables;

SHOW TABLES

Query id: 33ef7187-e01c-46c4-a968-53d9fbc1e691

┌─name─────────────────┐
│ columns_transformers │
│ insert_select_dst    │
│ insert_select_src    │
│ test_1               │
│ test_pk_order        │
└──────────────────────┘

5 rows in set. Elapsed: 0.003 sec.

dell123 :) select version();

SELECT version()

Query id: 487ee115-6dcb-4d54-a905-921e3ade99ae

┌─version()─┐
│ 20.12.1.1 │
└───────────┘

1 rows in set. Elapsed: 0.002 sec.

dell123 :) create table array(a Array(UInt8), b String) engine = Memory;

CREATE TABLE array
(
    `a` Array(UInt8),
    `b` String
)
ENGINE = Memory

Query id: 3376ee95-7dd9-4a99-816e-3c1ede3b99f2

Ok.

0 rows in set. Elapsed: 0.023 sec.

dell123 :) insert into array values([1:+1, 2], 'aaa');

INSERT INTO array VALUES

Query id: c0d95f63-128a-428b-84a9-8ae9da58077a

Error on processing query: insert into array values([1:+1, 2], 'aaa');
Code: 62, e.displayText() = DB::Exception: Cannot parse expression of type Array(UInt8) here: [1:+1, 2], 'aaa');: data for INSERT was parsed from query, Stack trace (when copying this message, always include the lines below):

0. /mnt/disk4/hewenting/ClickHouse/src/contrib/poco/Foundation/src/Exception.cpp:27: Poco::Exception::Exception(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int) @ 0x1189f3 in /mnt/disk4/hewenting/ClickHouse/build-dev_master/contrib/poco-cmake/Foundation/lib_poco_foundation.so
1. /mnt/disk4/hewenting/ClickHouse/src/src/Common/Exception.cpp:39: DB::Exception::Exception(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int) @ 0x1a3721 in /mnt/disk4/hewenting/ClickHouse/build-dev_master/src/libclickhouse_common_io.so
2. /mnt/disk4/hewenting/ClickHouse/src/contrib/libcxx/include/string:2134: DB::ValuesBlockInputFormat::parseExpression(DB::IColumn&, unsigned long) (.cold) @ 0x2b9cb2 in /mnt/disk4/hewenting/ClickHouse/build-dev_master/src/libclickhouse_processors_formats_impl.so
3. /mnt/disk4/hewenting/ClickHouse/src/src/Processors/Formats/Impl/ValuesBlockInputFormat.cpp:179: DB::ValuesBlockInputFormat::readRow(std::__1::vector<COW<DB::IColumn>::mutable_ptr<DB::IColumn>, std::__1::allocator<COW<DB::IColumn>::mutable_ptr<DB::IColumn> > >&, unsigned long) (.cold) @ 0x2ba1d4 in /mnt/disk4/hewenting/ClickHouse/build-dev_master/src/libclickhouse_processors_formats_impl.so
4. /mnt/disk4/hewenting/ClickHouse/src/src/Processors/Formats/Impl/ValuesBlockInputFormat.cpp:49: DB::ValuesBlockInputFormat::generate() @ 0x2b84cb in /mnt/disk4/hewenting/ClickHouse/build-dev_master/src/libclickhouse_processors_formats_impl.so
5. /mnt/disk4/hewenting/ClickHouse/src/contrib/libcxx/include/vector:1003: DB::ISource::work() @ 0x10cf67 in /mnt/disk4/hewenting/ClickHouse/build-dev_master/src/libclickhouse_processors.so
6. /mnt/disk4/hewenting/ClickHouse/src/src/Processors/Formats/InputStreamFromInputFormat.h:49: DB::InputStreamFromInputFormat::readImpl() @ 0x1a791d in /mnt/disk4/hewenting/ClickHouse/build-dev_master/src/libclickhouse_datastreams.so
7. /mnt/disk4/hewenting/ClickHouse/src/contrib/libcxx/include/vector:1003: DB::IBlockInputStream::read() @ 0x172cf3 in /mnt/disk4/hewenting/ClickHouse/build-dev_master/src/libclickhouse_datastreams.so
8. /mnt/disk4/hewenting/ClickHouse/src/contrib/libcxx/include/vector:1003: DB::AsynchronousBlockInputStream::calculate() @ 0x14a1ab in /mnt/disk4/hewenting/ClickHouse/build-dev_master/src/libclickhouse_datastreams.so
9. /mnt/disk4/hewenting/ClickHouse/src/contrib/libcxx/include/atomic:856: std::__1::__function::__func<DB::AsynchronousBlockInputStream::next()::'lambda'(), std::__1::allocator<DB::AsynchronousBlockInputStream::next()::'lambda'()>, void ()>::operator()() @ 0x14a649 in /mnt/disk4/hewenting/ClickHouse/build-dev_master/src/libclickhouse_datastreams.so
10. /mnt/disk4/hewenting/ClickHouse/src/contrib/libcxx/include/functional:1853: ThreadPoolImpl<ThreadFromGlobalPool>::worker(std::__1::__list_iterator<ThreadFromGlobalPool, void*>) @ 0x1f4e0c in /mnt/disk4/hewenting/ClickHouse/build-dev_master/src/libclickhouse_common_io.so
11. /mnt/disk4/hewenting/ClickHouse/src/src/Common/ThreadPool.h:177: ThreadFromGlobalPool::ThreadFromGlobalPool<void ThreadPoolImpl<ThreadFromGlobalPool>::scheduleImpl<void>(std::__1::function<void ()>, int, std::__1::optional<unsigned long>)::'lambda1'()>(void&&, void ThreadPoolImpl<ThreadFromGlobalPool>::scheduleImpl<void>(std::__1::function<void ()>, int, std::__1::optional<unsigned long>)::'lambda1'()&&...)::'lambda'()::operator()() @ 0x1f53f1 in /mnt/disk4/hewenting/ClickHouse/build-dev_master/src/libclickhouse_common_io.so
12. /mnt/disk4/hewenting/ClickHouse/src/contrib/libcxx/include/functional:1853: ThreadPoolImpl<std::__1::thread>::worker(std::__1::__list_iterator<std::__1::thread, void*>) @ 0x1f47cc in /mnt/disk4/hewenting/ClickHouse/build-dev_master/src/libclickhouse_common_io.so
13. /mnt/disk4/hewenting/ClickHouse/src/contrib/libcxx/include/memory:2615: void* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void ThreadPoolImpl<std::__1::thread>::scheduleImpl<void>(std::__1::function<void ()>, int, std::__1::optional<unsigned long>)::'lambda1'()> >(void*) @ 0x1f2d1f in /mnt/disk4/hewenting/ClickHouse/build-dev_master/src/libclickhouse_common_io.so
14. start_thread @ 0x7ef7 in /home/amos/gentoo/lib64/libpthread-2.31.so
15. __clone @ 0xf9bbf in /home/amos/gentoo/lib64/libc-2.31.so
 (version 2
dell123 :) show tables;

SHOW TABLES

Query id: bdbdd508-dc9b-4778-a7c4-fca290f3ae9c

Ok.

0 rows in set. Elapsed: 0.000 sec.

dell123 :) show tables;

SHOW TABLES

Query id: 46da7347-8959-421d-967c-6ca0c077695f

┌─name─────────────────┐
│ array                │
│ columns_transformers │
│ insert_select_dst    │
│ insert_select_src    │
│ test_1               │
│ test_pk_order        │
└──────────────────────┘

6 rows in set. Elapsed: 0.004 sec.

```


As we can see that the query "show tables" after "insert into xxx" shows nothing, But in fact there are several tables is this database.

**Comment by den-crane** [comment]
I able to reproduce it.

```
:) insert into array values([1:+1, 2], 'aaa');

INSERT INTO array VALUES

Query id: fa09aee8-3f4b-41f8-aae6-0272095bbb12

Error on processing query: insert into array values([1:+1, 2], 'aaa');
Code: 62, e.displayText() = DB::Exception: Cannot parse expression of type Array(UInt8) here: [1:+1, 2], 'aaa');: data for INSERT was parsed from query, Stack trace (when copying this message, always include the lines below):

0. DB::ValuesBlockInputFormat::parseExpression(DB::IColumn&, unsigned long) @ 0xe6a2335 in /usr/bin/clickhouse
1. DB::ValuesBlockInputFormat::readRow(std::__1::vector<COW<DB::IColumn>::mutable_ptr<DB::IColumn>, std::__1::allocator<COW<DB::IColumn>::mutable_ptr<DB::IColumn> > >&, unsigned long) @ 0xe69f80c in /usr/bin/clickhouse
2. DB::ValuesBlockInputFormat::generate() @ 0xe69ebd5 in /usr/bin/clickhouse
3. DB::ISource::work() @ 0xe58936a in /usr/bin/clickhouse
4. DB::InputStreamFromInputFormat::readImpl() @ 0xe55359f in /usr/bin/clickhouse
5. DB::IBlockInputStream::read() @ 0xd6cabb5 in /usr/bin/clickhouse
6. DB::AsynchronousBlockInputStream::calculate() @ 0xd6c6ffd in /usr/bin/clickhouse
7. ? @ 0xd6c7798 in /usr/bin/clickhouse
8. ThreadPoolImpl<ThreadFromGlobalPool>::worker(std::__1::__list_iterator<ThreadFromGlobalPool, void*>) @ 0x7d3b5ed in /usr/bin/clickhouse
9. ThreadFromGlobalPool::ThreadFromGlobalPool<void ThreadPoolImpl<ThreadFromGlobalPool>::scheduleImpl<void>(std::__1::function<void ()>, int, std::__1::optional<unsigned long>)::'lambda1'()>(void&&, void ThreadPoolImpl<ThreadFromGlobalPool>::scheduleImpl<void>(std::__1::function<void ()>, int, std::__1::optional<unsigned long>)::'lambda1'()&&...)::'lambda'()::operator()() @ 0x7d3db4f in /usr/bin/clickhouse
10. ThreadPoolImpl<std::__1::thread>::worker(std::__1::__list_iterator<std::__1::thread, void*>) @ 0x7d38aad in /usr/bin/clickhouse
11. ? @ 0x7d3c5d3 in /usr/bin/clickhouse
12. start_thread @ 0x7fa3 in /lib/x86_64-linux-gnu/libpthread-2.28.so
13. clone @ 0xf94cf in /lib/x86_64-linux-gnu/libc-2.28.so
 (version 20.12.1.5183 (offic
:) show tables;

SHOW TABLES

Query id: d3d7c899-e2d1-454c-a95d-a79d33ce34a5

Ok.

0 rows in set. Elapsed: 0.000 sec.

:) show tables;

SHOW TABLES

Query id: 893c2720-149e-4da9-b2dd-8b3697ed2fa6

┌─name──────┐
│ array     │
│ requests  │
│ test      │
│ test_join │
└───────────┘

4 rows in set. Elapsed: 0.001 sec.
```

exception spoils the next query.

```
2020.11.14 22:47:18.271955 [ 9076 ] {fa09aee8-3f4b-41f8-aae6-0272095bbb12} <Debug> executeQuery: (from [::1]:45216) insert into array values
2020.11.14 22:47:18.272024 [ 9076 ] {fa09aee8-3f4b-41f8-aae6-0272095bbb12} <Trace> ContextAccess (default): Access granted: INSERT(a, b) ON shard.array

2020.11.14 22:47:34.680435 [ 9076 ] {fa09aee8-3f4b-41f8-aae6-0272095bbb12} <Debug> MemoryTracker: Peak memory usage (for query): 0.00 B.
2020.11.14 22:47:34.680498 [ 9076 ] {} <Debug> TCPHandler: Processed in 16.408688953 sec.
2020.11.14 22:47:34.680610 [ 9076 ] {d3d7c899-e2d1-454c-a95d-a79d33ce34a5} <Debug> executeQuery: (from [::1]:45216) show tables;
2020.11.14 22:47:34.680705 [ 9076 ] {d3d7c899-e2d1-454c-a95d-a79d33ce34a5} <Debug> executeQuery: (internal) SELECT name FROM system.tables WHERE database = 'shard'
2020.11.14 22:47:34.680868 [ 9076 ] {d3d7c899-e2d1-454c-a95d-a79d33ce34a5} <Trace> ContextAccess (default): Access granted: SELECT(database, name) ON system.tables
2020.11.14 22:47:34.680951 [ 9076 ] {d3d7c899-e2d1-454c-a95d-a79d33ce34a5} <Trace> InterpreterSelectQuery: FetchColumns -> Complete
2020.11.14 22:47:34.681145 [ 9051 ] {d3d7c899-e2d1-454c-a95d-a79d33ce34a5} <Trace> ContextAccess (default): Access granted: SHOW TABLES ON *.*
2020.11.14 22:47:34.681262 [ 9076 ] {d3d7c899-e2d1-454c-a95d-a79d33ce34a5} <Information> executeQuery: Read 4 rows, 118.00 B in 0.000638526 sec., 6264 rows/sec., 180.47 KiB/sec.
2020.11.14 22:47:34.681271 [ 9076 ] {d3d7c899-e2d1-454c-a95d-a79d33ce34a5} <Debug> MemoryTracker: Peak memory usage (for query): 0.00 B.
2020.11.14 22:47:34.681302 [ 9076 ] {} <Debug> TCPHandler: Processed in 0.000784861 sec.

2020.11.14 22:47:38.731816 [ 9076 ] {} <Error> ServerErrorHandler: Code: 210, e.displayText() = DB::NetException: Connection reset by peer, while reading from socket ([::1]:45216), Stack trace (when copying this message, always include the lines below):

0. DB::ReadBufferFromPocoSocket::nextImpl() @ 0xda7898e in /usr/bin/clickhouse
1. DB::TCPHandler::runImpl() @ 0xe53faf4 in /usr/bin/clickhouse
2. DB::TCPHandler::run() @ 0xe54c707 in /usr/bin/clickhouse
3. Poco::Net::TCPServerConnection::start() @ 0x10cf7c4f in /usr/bin/clickhouse
4. Poco::Net::TCPServerDispatcher::run() @ 0x10cf965e in /usr/bin/clickhouse
5. Poco::PooledThread::run() @ 0x10e26e89 in /usr/bin/clickhouse
6. Poco::ThreadImpl::runnableEntry(void*) @ 0x10e22e1a in /usr/bin/clickhouse
7. start_thread @ 0x7fa3 in /lib/x86_64-linux-gnu/libpthread-2.28.so
8. clone @ 0xf94cf in /lib/x86_64-linux-gnu/libc-2.28.so
 (version 20.12.1.5183 (official build))

```


```sql
:) insert into array values([1:+1, 2], 'aaa');

INSERT INTO array VALUES

Query id: cf4f8a3e-4722-417f-ba3c-890599ab2711

Error on processing query: insert into array values([1:+1, 2], 'aaa');
Code: 62, e.displayText() = DB::Exception: Cannot parse expression of type Array(UInt8) here: [1:+1, 2], 'aaa');: data for INSERT was parsed from query, Stack trace (when copying this message, always include the lines below):


:) select version();

SELECT version()

Query id: a617467c-db1d-4322-a9c6-32e39f3d9416

Ok.

0 rows in set. Elapsed: 0.000 sec.

:) select version();

SELECT version()

Query id: 9436f6fb-0f3f-4164-97b2-8b48fc9bce4d

┌─version()────┐
│ 20.12.1.5183 │
└──────────────┘

1 rows in set. Elapsed: 0.001 sec.

```

**Comment by den-crane** [comment]
```
echo -e "insert into array values([1:+1, 2], 'aaa');\nselect version();"|clickhouse-client -nm -d shard --ignore-error

echo -e "insert into array values([1:+1, 2], 'aaa');\nselect version();\nselect version();"|clickhouse-client -nm -d shard --ignore-error
20.12.1.5183
```

**Comment by hexiaoting** [comment]
This bug is fixed in latest version: 21.3.1.1.
